File: 7-kyu/simple-fun-#149--next-day-of-week/javascript/solution.js
const nextDayOfWeek = (currentDay, availableWeekDays) => {
  const availableDays = availableWeekDays.toString(2).padStart(8, '0').split("").reverse().join("");
  let idx = currentDay % availableDays.length;
  while (availableDays[idx] !== '1') {
    idx++;
    idx = idx % availableDays.length;
  }
  return idx + 1;
};

File: 7-kyu/simple-fun-#149--next-day-of-week/python/solution.py
def next_day_of_week(current_day, available_week_days):
    avail_days = [ch for ch in f'{available_week_days:b}'.zfill(8)[::-1]]
    idx = current_day % len(avail_days)
    ret = avail_days[idx]
    while ret != '1':
        idx += 1
        idx = idx % len(avail_days)
        ret = avail_days[idx]
    return idx + 1
